lib/http_get.js

```diff
@@ -0,0 +1,312 @@
+ * @param {Number} chunkId - Current chunk number
+ * @param {Number} fragmenId - Current fragment number
+ * @param {Number} requestTimeoutMs - Timeout of each sub-query
+ * @param {null|[Number]} range - HTTP range requested
+ * @return {undefined}
+ */
+function fragmentGET(
+    { httpAgent, errorAgent, opContext, callback,
+      chunkId, fragmentId, requestTimeoutMs, range }) {
+    const isData = fragmentId < opContext.fragments.nDataParts;
+    const chunk = opContext.fragments.chunks[chunkId];
+    const fragment = isData ? chunk.data[fragmentId] :
+              chunk.coding[fragmentId - opContext.fragments.nDataParts];
+
+    const reqContext = { opContext, chunkId, fragmentId };
+    const { hostname, port, key } = fragment;
+    const requestOptions = httpUtils.getCommonStoreRequestOptions(
+        httpAgent, hostname, port, key);
+
+    requestOptions.method = 'GET';
+    requestOptions.headers = {
+        ['Accept']: protocol.helpers.makeAccept(
+            ['data', range], ['crc']),
+    };
+
+    const request = httpUtils.newRequest(
+        requestOptions, opContext.log, reqContext,
+        requestTimeoutMs,
+        /* callback */
+        reqCtx => _fragmentGETCb(reqCtx, callback, errorAgent));
+
+    request.end();
+}
+
+
+/* eslint-disable no-unused-vars */
+function _getChunkRange(fragments, chunkId, globalRange) {
+    return globalRange;
+}
+/* eslint-enable no-unused-vars */
+
+/**
+ * Get all fragments of an object
+ *
+ * @param {http.HttpAgent} httpAgent - Agent to use
+ * @param {Object} errorAgent - Agent to persist errors
+ * @param {werelogs.Logger} logger - Logger to use
+ * @param {Object} fragments - Object description to delete
+ * @param {String} rawKey - Uri of the object
+ *                 (refer to keyscheme.js for content)
+ * @param {HyperdriveClient~getCallback} callback - Callback
+ * @param {Number} requestTimeoutMs - Timeout of each sub-query
+ * @param {Number [] | Undefined} range - Range (if any) with
+ *                                        first element the start
+ *                                        and the second element the end
+ * @returns {Object} Operation context tracking everything
+ */
+function doGET({ httpAgent, errorAgent, log, fragments, rawKey,
+                 callback, requestTimeoutMs, range }) {
+    // Split, replication or erasure coding is currently not supported
+    assert.strictEqual(fragments.nChunks, 1);
+    assert.strictEqual(fragments.nCodingParts, 0);
+
+    const opContext = httpUtils.makeOperationContext(fragments, rawKey, log);
+
+    /*
+     * TODO: split? Request all? First 2 then the next when
+     *       first is consumed? Introduced a small delay (1ms)?
+     * TODO: handle range on split
+     * TODO: erasure coding
+     */
+    split.getChunkSlice(fragments, range).forEach((chunk, chunkId) => {
+        const chunkRange = _getChunkRange(fragments, chunkId, range);
+        chunk.data.forEach((fragment, fragmentId) => {
```

Comment:
Now I'm confused about what's a split, what's a chunk, and what's a fragment ^^'

Reply:
split is the name of a module (would probably be best to uppercase or rename it?). split.getChunkSlice() (will) handle range queries over a splitted: essentially only returning a slice of chunks to retrieved.

A splitted object is made of several chunks (ordered by offset). Each chunk is made of several fragments (replicated, erasure coded, etc). I tried my best to keep the naming consistent across all operations and Design.md but some may have slipped through the cracks